client/utils/charge/index.js

```diff
@@ -60,27 +60,53 @@ export const getChargeStatus = ( charge = {} ) => {
 };
 
 /**
- * Calculates display values for charge amounts.
+ * Calculates display values for charge amounts in settlement currency.
  *
  * @param {Object} charge The full charge object.
- * @return {Object} An object, containing the `net`, `fee`, and `refund` amounts in Stripe format (*100).
+ * @return {Object} An object, containing the `currency`, `amount`, `net`, `fee`, and `refunded` amounts in Stripe format (*100).
  */
 export const getChargeAmounts = ( charge ) => {
-	// The base fee is the application fee.
-	let fee = charge.application_fee_amount;
-	let refunded = 0;
+	const balance = {
+		currency: charge.currency || 'USD',
+		amount: charge.amount,
+		fee: charge.application_fee_amount,
+		refunded: 0,
+		net: 0,
+	};
 
-	if ( isChargeDisputed( charge ) ) {
-		fee += sumBy( charge.dispute.balance_transactions, 'fee' );
-		refunded -= sumBy( charge.dispute.balance_transactions, 'amount' );
+	if ( isChargeRefunded( charge ) ) {
+		balance.refunded += charge.amount_refunded;
 	}
 
-	if ( isChargeRefunded( charge ) ) {
-		refunded += charge.amount_refunded;
+	if (
+		isObject( charge.balance_transaction ) &&
+		charge.balance_transaction.currency !== charge.currency
+	) {
+		balance.currency = charge.balance_transaction.currency;
+		balance.amount = charge.balance_transaction.amount;
+		balance.fee = charge.balance_transaction.fee;
+		balance.refunded = 0;
+
+		if ( isChargeRefunded( charge ) ) {
+			// Refund balance_transactions have negative amount.
+			balance.refunded -= sumBy(
+				charge.refunds.data,
+				'balance_transaction.amount'
+			);
+		}
```

Comment:
Consider moving those calculations to PHP as suggested in 528-gh-Automattic/woocommerce-payments-server#discussion_r564417133.